coral-component-steplist/src/scripts/StepList.js

```diff
@@ -227,26 +258,35 @@ class StepList extends BaseComponent(HTMLElement) {
   
     const isInteractive = this._interaction === interaction.ON;
     this.classList.toggle(`${CLASSNAME}--interactive`, isInteractive);
-  
-    const steps = this.items.getAll();
-    const stepsCount = steps.length;
+
+    if (!this.items.length) {
+      return;
+    }
   
     // update tab index for all children
-    for (let i = 0; i < stepsCount; i++) {
-      this._syncItemTabIndex(steps[i]);
+    const _syncItemProps = () => {
+      const steps = this.items.getAll();
+      const stepsCount = steps.length;
+
+      for (let i = 0; i < stepsCount; i++) {
+        // update tab index for all children
+        steps[i]._syncTabIndex(isInteractive);
+        //update posin set and total size for all steps
+        steps[i]._syncSizeAndCurrentIndex(i + 1, stepsCount);
+      }
+    };
+
+    if (typeof this.items.last()._syncTabIndex === 'function') {
```

Comment:
You can store `this.items.last()` in a variable to avoid calling it twice if the first condition is false.